### ISSUE TITLE
[FEA] Remove from_xxx methods in DataFrame interface, or at least add other datatype sources to DataFrame object

### ISSUE THREAD

**Issue by miguelusque** [rapidsai/cudf#5277]
**Is your feature request related to a problem? Please describe.**
Hi,

I was wondering if, instead of providing the following methods, it would be better to accept different datatypes when creating a DataFrame:
 
  - DataFrame.from_pandas
  - DataFrame.from_records
  - DataFrame.from_gpu_matrix
  - DataFrame.from_arrow

I think it would make easier to create DataFrames by allowing these datatypes as input when creating a DataFrame, instead of having specific methods. 

In the future, if other datatypes are supported, the code would remain valid without any change.

I am looking forward for your comments on this.

Regards,
Miguel

**Comment by jl-massey** [comment]
I think the main driver is to try to stay consistent with the look and feel of Pandas.

**Comment by miguelusque** [comment]
Hi @jl-massey ,

IIRC, Pandas has from_records method, but none of the other methods detailed in my comment.

IMO, It might make sense to keep from_records for compatibility purposes, but a different approach could be considered for the other sources of data.

**Comment by jl-massey** [comment]
@miguelusque Fair enough, pandas uses from_<format> or read_<format>.   I expect, with the broad range of source types and handling, that your request would 'practically' be added as a method that calls one of the others after type inference?  i.e. read_type("some_file.csv.gz") --> read_csv("some_file.csv.gz") ??

**Comment by miguelusque** [comment]
I have just double-checked this feature request, and the only missing datatype from the list above to be supported by `cudf.DataFrame()` is  `cudf.DataFrame.from_arrow()`.

Could you please consider adding support to that method too? Thanks!

**Comment by shwina** [comment]
I think ultimately we want to deprecate/remove the `.from_pandas()` API and be able to support constructing from other "DataFrameLike" objects (e.g., Pandas, Modin, Vaex) via the `.from_dataframe()` API. Currently, that depends on https://github.com/rapidsai/cudf/pull/11392. 

As for `.from_arrow()`, I'd rather we have a more _explicit_ `.from_arrow` constructor than an _implicit_ one that goes through the constructor. I understand that Pandas does not have the equivalent `.from_arrow()` method, but neither does it allow passing pyarrow tables as input to the `DataFrame` constructor (AFAICT). What do you think?

**Comment by vyasr** [comment]
As of today (pandas 2.3.0 and cudf 25.08 on my machine), we have the following `Dataframe` factories:
- `from_dict`: Shared with pandas
- `from_records`: Shared with pandas
- `from_pylibcudf`: Specific to cudf, must exist
- `from_pandas`: Based on some local testing, this function is equivalent to what we already support via the constructor and has comparable performance
```
In [6]: pdf = cudf.datasets.randomdata().to_pandas()

In [7]: %timeit cudf.DataFrame(pdf)
1.9 ms ± 7.73 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [8]: %timeit cudf.DataFrame.from_pandas(pdf)
1.87 ms ± 8.61 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [11]: for _ in range(10):
    ...:     pdf = cudf.datasets.randomdata(dtypes={"id": int, "x": float, "y": str, "z": "category"}).to_pandas()
    ...:     cudf.DataFrame.from_pandas(pdf)
    ...:     cudf.DataFrame(pdf)
    ...:
```
- `from_arrow`: This is the one case where I could see an argument for adding support. The problem is that pandas does not support this either. 
```
In [16]: pd.DataFrame(pa.Table.from_pandas(pdf))
...
ValueError: DataFrame constructor not properly called!
```
The [pandas documentation](https://pandas.pydata.org/docs/user_guide/pyarrow.html) indicates that the expected way to perform this conversion is in the other direction:
> To convert a [pyarrow.Table](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table) to a [DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame), you can call the [pyarrow.Table.to_pandas()](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.to_pandas) method with types_mapper=pd.ArrowDtype.

As such, we would be deviating further from pandas behavior by supporting this in a preexisting API rather than maintaining the existing one.

**Conclusions**
I am inclined to leave arrow support out of the constructor, deprecate `from_pandas`, and consider this issue closed. I could also be convinced to add `from_arrow` support to the constructor.

**Comment by mroeschke** [comment]
Although I like the explicitness of `from_pandas`, I would be OK with deprecating it.

A natural follow up is if we also deprecate `.from_pandas` class methods from other objects like `Series`,`Index`, etc. 

The only _minor_ inconveniences  is that `cudf.MultiIndex(pd.MultiIndex, ...)` isn't a valid construction unlike `cudf.Series(pd.Series)` or `cudf.Index(pd.Index)` so the user would need to do a little more unraveling for the `MultiIndex` case (i.e. pass the levels and codes manually).

**Comment by vyasr** [comment]
The alternative would be to drop support for pandas objects in the constructor and instead require users to use `from_pandas`. My main concern with that is that it might break a lot of code that's already out there.

**Comment by vyasr** [comment]
This was resolved in #19996. It didn't autoclose because I merged into 25.12, which isn't yet the default branch.
